src/barChart.ts

```diff
@@ -347,38 +344,53 @@ export class BarChart implements IVisual {
         // Turn on landing page in capabilities and remove comment to turn on landing page!
         // this.HandleLandingPage(options);
+        this.formattingSettings = this.formattingSettingsService.populateFormattingSettingsModel(BarChartSettingsModel, options.dataViews?.[0]);
+        this.barDataPoints = createSelectorDataPoints(options, this.host);
+        this.formattingSettings.populateColorSelector(this.barDataPoints);
+        this.formatMode = options.formatMode;
         const width = options.viewport.width;
         let height = options.viewport.height;
 
         this.svg
             .attr("width", width)
             .attr("height", height);
 
-        if (settings.enableAxis.show) {
+        if (this.formattingSettings.enableAxis.show.value) {
             const margins = BarChart.Config.margins;
             height -= margins.bottom;
         }
 
         this.helpLinkElement
-            .classed("hidden", !settings.generalView.showHelpLink)
-            .style("border-color", settings.generalView.helpLinkColor)
-            .style("color", settings.generalView.helpLinkColor);
+            .classed("hidden", !this.formattingSettings.generalView.showHelpLink.value)
+            .style("border-color", this.formattingSettings.generalView.helpLinkColor)
+            .style("color", this.formattingSettings.generalView.helpLinkColor);
+
+        this.directEditElement
+            .classed('direct-edit', true)
+            .classed('hidden', !this.formattingSettings.directEditSettings.show.value)
+            .attr(SubSelectableObjectNameAttribute, 'directEditTest')
+            .attr(SubSelectableDisplayNameAttribute, 'Direct Edit')
+            .attr(SubSelectableDirectEditAttr, this.visualDirectEditSubSelection)
+            .classed(HtmlSubSelectableClass, options.formatMode && this.formattingSettings.directEditSettings.show.value)
+            .style('font-family', this.formattingSettings.directEditSettings.font.fontFamily.value)
+            .style('color', this.formattingSettings.directEditSettings.fontColor.value.value)
+            .style('font-style', this.formattingSettings.directEditSettings.font.italic.value ? 'italic' : 'normal')
+            .style('text-decoration', this.formattingSettings.directEditSettings.font.underline.value ? 'underline' : 'none')
+            .style('font-weight', this.formattingSettings.directEditSettings.font.bold.value ? 'bold' : 'normal')
+            .text(this.formattingSettings.directEditSettings.textProperty.value);
 
         this.xAxis
             .style("font-size", Math.min(height, width) * BarChart.Config.xAxisFontMultiplier)
-            .style("fill", settings.enableAxis.fill);
+            .style("fill", this.formattingSettings.enableAxis.fill.value.value);
 
         const yScale = scaleLinear()
-            .domain([0, viewModel.dataMax])
+            .domain([0, <number>options.dataViews[0].categorical.values[0].maxLocal])
```

Comment:
why  maxLocal?  
Plus why its taken from options object not from formattingSettings

Reply:
this is from the values and not metadata objects